src/node_os.cc

```diff
@@ -87,29 +87,41 @@ static void GetOSType(const FunctionCallbackInfo<Value>& args) {
     return args.GetReturnValue().SetUndefined();
   }
 
-  args.GetReturnValue().Set(
-      String::NewFromUtf8(env->isolate(), info.sysname, NewStringType::kNormal)
-          .ToLocalChecked());
+  Local<Value> version_str = String::NewFromUtf8(env->isolate(),
+                              info.version,
+                              v8::NewStringType::kNormal).ToLocalChecked();
+
+  Local<Value> release_str = String::NewFromUtf8(env->isolate(),
+                              info.release,
+                              v8::NewStringType::kNormal).ToLocalChecked();
+
+  Local<Value> sysname_str = String::NewFromUtf8(env->isolate(),
+                              info.sysname,
+                              v8::NewStringType::kNormal).ToLocalChecked();
+
+  Local<Object> obj = Object::New(env->isolate());
+  obj->Set(context,
+           String::NewFromUtf8(env->isolate(),
+                               "type",
+                               NewStringType::kNormal).ToLocalChecked(),
+                               sysname_str)
+           .FromJust();
+  obj->Set(context,
+           String::NewFromUtf8(env->isolate(),
+                               "version",
+                               NewStringType::kNormal).ToLocalChecked(),
+                               version_str)
+           .FromJust();
+  obj->Set(context,
+           String::NewFromUtf8(env->isolate(),
+                               "release",
+                               NewStringType::kNormal).ToLocalChecked(),
+                               release_str)
+           .FromJust();
```

Comment:
It might be nicer to create an Array with 3 entries here, even if that means that means that the order of the entries becomes important:

1. It removes the need to create strings from the keys in C++, and
2. The `.FromJust()` calls here can be brought to crash if userland code runs before the `os` module is required

E.g. the following would current crash the process:

```
$ node -e 'Object.defineProperty(Object.prototype, "release", { set() { throw new Error(); } }); require("os")'
FATAL ERROR: v8::FromJust Maybe value is Nothing.
 1: 0xa15eb0 node::Abort() [./node]
```

Reply:
Fwiw, here’s a good example of how to create a JS Array from C++ without much overhead:

https://github.com/nodejs/node/blob/d63bcdd9cd81b10b6158f40d44bf6f37e5f54385/src/node_util.cc#L100-L106

Reply:
> E.g. the following would current crash the process:

Are these kind of crashes something to be avoided? It can be reproduced with some other properties:
```sh
node -e 'Object.defineProperty(Object.prototype, "name", { set() { throw new Error(); } });require("dns").resolveSrv("_jabber._tcp.google.com", () => {});'
```

Reply:
@santigimeno Yes – unfortunately, a lot of Node.js core C++ code doesn’t do error handling well, but new code generally should.